ethp2p/rlpx.nim

```diff
@@ -215,9 +212,11 @@ proc dispatchMsg(peer: Peer, msgId: int, msgData: var Rlp) =
 
   thunk(peer, msgData)
 
-proc send(p: Peer, data: BytesRange): Future[void] =
+proc send(p: Peer, data: BytesRange) {.async.} =
   var cipherText = encryptMsg(data, p.secretsState)
+  GC_ref(cipherText)
```

Comment:
Since, there is no waiting in this `async` proc, are these `GC_ref/unref` calls actually achieving anything?

Reply:
Good catch. Fixed.

Reply:
Yeah, I know... But i'm thinking about fixing this eventually, so that local closure iter vars will be stack-based when referred from a single state. And `GC_ref/GC_unref` will be required in such cases. So thats just future-proof...

Reply:
Actually this is bad idea, passing stack variables... and we already talked about it (and you already has this mistake in `encryptMsg`)